numpy/lib/histograms.py

```diff
@@ -219,18 +219,21 @@ def _get_outer_edges(a, range):
     """
     if range is not None:
         first_edge, last_edge = range
+        if not (np.isfinite(first_edge) and np.isfinite(last_edge)):
+            raise ValueError(
+                "supplied range of [{}, {}] is not finite".format(first_edge, last_edge))
+        if first_edge > last_edge:
+            raise ValueError(
+                'max must be larger than min in range parameter.')
     elif a.size == 0:
         # handle empty arrays. Can't determine range, so use 0-1.
         first_edge, last_edge = 0, 1
     else:
         first_edge, last_edge = a.min(), a.max()
 
-    if first_edge > last_edge:
-        raise ValueError(
-            'max must be larger than min in range parameter.')
     if not (np.isfinite(first_edge) and np.isfinite(last_edge)):
```

Comment:
This can move into the `else` above too now

Reply:
I'm referring to the isfinire checks here, not the change above

Reply:
Oh alright. My bad, I've made the change.